File: scala-3.7-spring-boot-3.5-netty-metrics-grafana-k6/src/main/scala/com/github/diegopacheco/scala3/sb3/model/DataEntry.scala
package com.github.diegopacheco.scala3.sb3.model

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

import java.time.LocalDateTime
import scala.beans.BeanProperty

@Table("data_entries")
case class DataEntry(
  @Id @BeanProperty var id: java.lang.Long,
  @BeanProperty var name: String,
  @BeanProperty var value: String,
  @BeanProperty var createdAt: LocalDateTime
) {
  def this() = this(null, null, null, null)
}

File: scala-3.7-spring-boot-3.5-netty-multi-workers-metrics-grafana-k6/core/src/main/java/com/github/diegopacheco/scala3/sb3/core/model/DataEntry.java
package com.github.diegopacheco.scala3.sb3.core.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Table("data_entries")
public class DataEntry {
    @Id
    private Long id;
    private String name;
    private String value;
    private LocalDateTime createdAt;

    public DataEntry() {
    }

    public DataEntry(Long id, String name, String value, LocalDateTime createdAt) {
        this.id = id;
        this.name = name;
        this.value = value;
        this.createdAt = createdAt;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}
